templates/app/angular-default/hooks/post-create.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const spawn = require('child_process').spawn; // eslint-disable-line security/detect-child-process
+
+exports.id = 'com.appcelerator.angular.post-create';
+
+exports.init = (logger, config, cli) => {
+	cli.on('create.post.app', (creator, next) => {
+		const projectName = cli.argv.name;
+		const projectPath = path.join(cli.argv['workspace-dir'], projectName);
+		const packageJsonPath = path.join(projectPath, 'app', 'package.json');
+		if (fs.existsSync(packageJsonPath)) {
+			let packageJson = JSON.parse(fs.readFileSync(packageJsonPath).toString());
+			packageJson.name = projectName;
+			fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 4));
+		}
+
+		logger.info('Installing Angular project dependencies');
+		let errorOutput = '';
+		const child = spawn('npm', [ 'i' ], { cwd: path.dirname(packageJsonPath) });
```

Comment:
Executing npm (or similar) commands through spawn is problematic, usually ending up with an ENOENT. I think one of the following should suffice (personally I think `shell: true`)

* Pass `shell: true` in the options of the spawn call. Only supported on Node 5.7.0 and up and we do support Node 4, but looking at the code I think it's safe to maybe error if you're using less than Node 8 or 6?
* Use exec instead of spawn (I think). But might run into maxbuffer issues
* Do the same as `shell: true` manually. Something like `spawn('cmd', ['/s', '/c', 'npm', 'i'],{ cwd: path.dirname(packageJsonPath) });`